Modules/HRV_DFA/DFA.cpp

```diff
@@ -0,0 +1,332 @@
+#include "DFA.h"
+
+using namespace std;
+
+// creating array with natural numbers for dfa algorhitm
+vector<int> inputData::createNarray(int start, int end)
+{
+    vector <int> n;
+    int size = end-start+1;
+    for(int i = 0; i<size; i++)
+    {
+        n.push_back(i+start);
+    }
+
+    return n;
+}
+
+// create vector with proper data
+// input comes from higher level module as number of R-peak sample
+vector<long double> inputData::readInput(vector<int>& input, int fs)
+{
+    // this only for debugging
+    // delete before use in application
+    ifstream ifile ("C://Users//Natalia//Desktop//dadm projekt//ECG-Analyzer//Modules//HRV_DFA//nsr001.txt");
+    double num = 0.0;
+
+    if (!ifile.is_open()) {
+            std::cerr << "There was a problem opening the input file!\n";
+            exit(1);//exit or do additional error checking
+        }
+
+    while (ifile >> num) {
+            data.push_back(num);
+        };
+
+    // for proper application
+    // uncomment this before use in application
+//    int len = int (input.size());
+//    int k = 0;
+//    int temp;
+
+//    while (k < len-1)
+//    {
+//        temp = input.at(k+1) - input.at(k);
+//        data.push_back( temp/fs );
+//        k++;
+//    }
+
+    length = int(data.size());
+
+    for(int i = 0; i<length; i++)
+        xmean += data.at(i);
+
+    xmean = xmean/length;
+
+    for( size_t i = 0; i < data.size(); i++ )
```

Comment:
Pierwsza sprawa to: postarajmy się maksymalnie zmniejszyć liczbę operacji kopiowania i przypisania wartości wektorów. Dane wczytane mają znaczącą liczbę rekordów (setki tysięcy) więc każda operacja kopiowania znacząco odbije się nam na wydajności. Druga: nie używajmy pętli w takiej formie - `std::for_each` lub range-based loop. Zastosuj proszę te uwagi do całego swojego kodu

Reply:
do poprawy w następnych commitach